fw_103/src/main.c

```diff
@@ -0,0 +1,73 @@
+#include <stdbool.h>
+#include <stdint.h>
+#include "adc.h"
+#include "delay.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "interrupt_def.h"
+#include "log.h"
+#include "status.h"
+#include "wait.h"
+
+static void prv_button_interrupt_handler(const GpioAddress *address, void *context) {
+  GpioAddress *adc_address = (GpioAddress *)context;
+  uint16_t adc_data = 0;
+  adc_read_converted_pin(*(adc_address), &adc_data);
+  LOG_DEBUG("ADC Data: %d\n", adc_data);
+  delay_ms(200);
+}
+
+int main(void) {
+  interrupt_init();
+  gpio_init();
+  gpio_it_init();
+
+  static GpioAddress button_addr = {

```

Comment:
static declarations should only be used outside of functions, unless you have a good reason for using it in a function